cmake/Modules_CUDA_fix/upstream/FindCUDAToolkit.cmake

```diff
@@ -1035,38 +1057,48 @@ if(CUDAToolkit_FOUND)
   endif()
 
   _CUDAToolkit_find_and_add_import_lib(nvrtc DEPS cuda_driver)
+  if(CUDAToolkit_VERSION VERSION_GREATER_EQUAL 11.1.0)
+    if(NOT TARGET CUDA::nvptxcompiler_static)
+      _CUDAToolkit_find_and_add_import_lib(nvptxcompiler_static DEPS cuda_driver)
+      if(TARGET CUDA::nvptxcompiler_static)
+        target_link_libraries(CUDA::nvptxcompiler_static INTERFACE Threads::Threads)
+      endif()
+    endif()
+  endif()
 
   _CUDAToolkit_find_and_add_import_lib(nvml ALT nvidia-ml nvml)
 
-  # nvtools can be installed outside the CUDA toolkit directory,
-  # so search the NVTOOLSEXT_PATH windows only environment variable
-  set(nvToolsExt_EXTRA_PATHS)
   if(WIN32)
-     set(nvToolsExt_EXTRA_PATHS
-         "$ENV{NVTOOLSEXT_PATH}"
-         "C:\\Program Files\\NVIDIA Corporation\\NvToolsExt")
+    # nvtools can be installed outside the CUDA toolkit directory
+    # so prefer the NVTOOLSEXT_PATH windows only environment variable
+    # In addition on windows the most common name is nvToolsExt64_1
+    find_library(CUDA_nvToolsExt_LIBRARY
+      NAMES nvToolsExt64_1 nvToolsExt64 nvToolsExt
+      PATHS ENV NVTOOLSEXT_PATH
+            ENV CUDA_PATH
+      PATH_SUFFIXES lib/x64 lib
+    )
   endif()
+  _CUDAToolkit_find_and_add_import_lib(nvToolsExt ALT nvToolsExt64)
 
-  find_path(CUDAToolkit_nvToolsExt_INCLUDE_DIR nvToolsExt.h
-      PATHS "${CUDAToolkit_INCLUDE_DIR}"
-            "${CUDAToolkit_ROOT_DIR}"
-            ${nvToolsExt_EXTRA_PATHS}
-      PATH_SUFFIXES include
-      NO_DEFAULT_PATH)
-  mark_as_advanced(CUDAToolkit_nvToolsExt_INCLUDE_DIR)
+  if(CUDAToolkit_VERSION VERSION_GREATER_EQUAL 10.0)
+    # nvToolsExt is deprecated since nvtx3 introduction.
+    # Warn only if the project requires a sufficiently new CMake to make migration possible.
+    if(TARGET CUDA::nvToolsExt AND CMAKE_MINIMUM_REQUIRED_VERSION VERSION_GREATER_EQUAL 3.25)
+      set_property(TARGET CUDA::nvToolsExt PROPERTY DEPRECATION "nvToolsExt has been superseded by nvtx3 since CUDA 10.0 and CMake 3.25. Use CUDA::nvtx3 and include <nvtx3/nvToolsExt.h> instead.")
+    endif()
 
-  if(CUDAToolkit_nvToolsExt_INCLUDE_DIR)
-    _CUDAToolkit_find_and_add_import_lib(nvToolsExt
-        ALT nvToolsExt64 nvToolsExt64_1
-        EXTRA_HINTS ${nvToolsExt_EXTRA_PATHS}
-        EXTRA_INCLUDE_DIRS "${CUDAToolkit_nvToolsExt_INCLUDE_DIR}")
```

Comment:
Windows failure is real, my version of `FindCUDAToolkit.cmake` has fixes not in the upstream module, including the `EXTRA_HINTS` argument here.

```
C:\actions-runner\_work\pytorch\pytorch\torch\csrc\profiler\stubs\cuda.cpp(3): fatal error C1083: Cannot open include file: 'nvToolsExt.h': No such file or directory
```

Reply:
OK, revert to your variant